src/components/cc-token-oauth-list/cc-token-oauth-list.smart.js

```diff
@@ -0,0 +1,223 @@
+// prettier-ignore
+// @ts-expect-error FIXME: remove when clever-client exports types
+import { todo_listSelfTokens as getAllTokens,todo_revokeSelfToken as revokeToken } from '@clevercloud/client/esm/api/v2/user.js';
+import { notifyError, notifySuccess } from '../../lib/notifications.js';
+import { sendToApi } from '../../lib/send-to-api.js';
+import { defineSmartComponent } from '../../lib/smart/define-smart-component.js';
+import { i18n } from '../../translations/translation.js';
+import '../cc-smart-container/cc-smart-container.js';
+import './cc-token-oauth-list.js';
+
+/**
+ * @typedef {import('./cc-token-oauth-list.js').CcTokenOauthList} CcTokenOauthList
+ * @typedef {import('./cc-token-oauth-list.types.js').OauthTokenState} OauthTokenState
+ * @typedef {import('./cc-token-oauth-list.types.js').TokenOauthListStateLoaded} TokenOauthListStateLoaded
+ * @typedef {import('./cc-token-oauth-list.types.js').TokenOauthListStateRevokingAll} TokenOauthListStateRevokingAll
+ * @typedef {import('./cc-token-oauth-list.types.js').OauthTokenStateRevoking} OauthTokenStateRevoking
+ * @typedef {import('./cc-token-oauth-list.types.js').OauthToken} OauthToken
+ * @typedef {import('../cc-token-session-list/cc-token-session-list.types.js').RawTokenData} RawTokenData
+ * @typedef {import('../../lib/send-to-api.types.js').ApiConfig} ApiConfig
+ * @typedef {import('../../lib/smart/smart-component.types.js').OnContextUpdateArgs<CcTokenOauthList>} OnContextUpdateArgs
+ */
+
+defineSmartComponent({
+  selector: 'cc-token-oauth-list',
+  params: {
+    apiConfig: { type: Object },
+    authBridgeConsumerKey: { type: String },
+  },
+  /** @param {OnContextUpdateArgs} args */
+  onContextUpdate({ component, context, onEvent, updateComponent }) {
+    const { apiConfig, authBridgeConsumerKey } = context;
+    const api = new Api(apiConfig, authBridgeConsumerKey);
+
+    /**
+     * Updates a single OAuth token
+     *
+     * @param {string} tokenId The ID of the token to update
+     * @param {function(OauthTokenState): void} callback A callback function to execute with the updated token
+     */
+    function updateOneToken(tokenId, callback) {
+      updateComponent(
+        'state',
+        /** @param {TokenOauthListStateLoaded} state */
+        (state) => {
+          const oauthTokenToUpdate = state.oauthTokens.find((token) => token.id === tokenId);
+
+          if (oauthTokenToUpdate != null) {
+            callback(oauthTokenToUpdate);
+          }
+        },
+      );
+    }
+
+    updateComponent('state', { type: 'loading' });
+
+    api
+      .getOauthTokens()
+      .then((tokens) => {
+        const formattedTokens = tokens.map(
```

Comment:
nitpick: I don't really like `formatted` (I don't see any formatting process here). we can rename it to `oauthTokens` or `oauthTokensStates`